.github/workflows/ensnode_railway_deploy.yml

```
@@ -0,0 +1,111 @@
+name: ENSNode - Build & push ENSNode Docker image (and conditionally deploy to Railway)
+
+on:
+  push:
+    branches:
+      # Build and push 'latest' Docker image. Skip deployments
+      - main
+      # Build and push 'alpha' Docker image. Update 'alpha' Railway shared environment variables.
+      - alpha
+      # Build and push 'subgraph' Docker image. Update 'subgraph' Railway shared environment variables.
+      - subgraph
+      # More about `Railway shared variables': https://docs.railway.com/guides/variables#shared-variables
+
+jobs:
+  image-build-and-push:
+    runs-on: ubuntu-latest
+    env:
+      RAILWAY_TOKEN: ${{ secrets.RAILWAY_TOKEN }}
+      # Token for redeploy. Redeploy needs a separate team token. 
+      RAILWAY_TEAM_TOKEN: ${{ secrets.RAILWAY_TEAM_TOKEN }}
+      RAILWAY_PROJECT_ID: ${{ secrets.RAILWAY_PROJECT_ID }}
+      RAILWAY_ENVIRONMENT_ID: ${{ secrets.RAILWAY_ENVIRONMENT_ID}}
+      # For now just use github.sha. Should be extended in the future
+      DATABASE_SCHEMA: ${{ github.sha }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Extract Branch Name
+        run: echo "BRANCH=${GITHUB_REF#refs/heads/}" >> $GITHUB_ENV
+    
+      # We need to recognize which branch had an incoming push
+      # Based on that build a Docker image with corresponding tag - use the TAG variable for this
+      # For the alpha/subgraph TAG we also need to specify the corresponding Railway SVC_ID which is used for service redeployment
+      - name: Determine Docker image tag
+        run: |
+            case "$BRANCH" in
+              "main")
+                echo "TAG=latest" >> $GITHUB_ENV
+                ;;
+              "alpha")
+                echo "TAG=alpha" >> $GITHUB_ENV
+                echo "SVC_ID="${{ secrets.ALPHA_SVC_ID }} >> "$GITHUB_ENV"
+                ;;
+              "subgraph")
+                echo "TAG=subgraph" >> $GITHUB_ENV
+                echo "SVC_ID="${{ secrets.SUBGRAPH_SVC_ID }} >> "$GITHUB_ENV"
+                ;;
+              *)
+                echo "Branch not recognized, skipping workflow"
+                exit 1
+            esac
+
+      - name: Build & Push Docker image for ENSNode to Github Container registry
+        env:
+          IMAGE_NAME: ghcr.io/${{ github.repository }}/ensnode
```

```suggestion
          IMAGE_NAME: ghcr.io/${{ github.repository }}/ensindexer
```

Is that right? Please fix as needed.